src/main/scala/com/gu/tip/assertion/TipAssert.scala

```diff
@@ -0,0 +1,52 @@
+package com.gu.tip.assertion
+
+import java.util.concurrent.Executors
+
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+
+import scala.concurrent.duration._
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Random, Success, Try}
+
+sealed trait AssertionResult
+case object TipAssertPass extends AssertionResult
+case object TipAssertFail extends AssertionResult
+
+trait TipAssertIf {
+  protected val logger: Logger
+
+  val threadPool = Executors.newFixedThreadPool(
+    2,
+    (r: Runnable) =>
+      new Thread(r, s"tip-assertion-pool-thread-${Random.nextInt(2)}"))
```

Comment:
I assume the closing ) is supposed to be lined up
```suggestion
      new Thread(r, s"tip-assertion-pool-thread-${Random.nextInt(2)}")
  )
```